htdocs/core/class/commondocgenerator.class.php

```diff
@@ -540,10 +540,11 @@ function get_substitutionarray_lines($line,$outputlangs)
 		// Load product data optional fields to the line -> enables to use "line_options_{extrafield}"
 		if (isset($line->fk_product) && $line->fk_product > 0)
 		{
+			$extrafields = new ExtraFields($this->db);
+			$extralabels = $extrafields->fetch_name_optionals_label($extrafieldkey,true);
 			$tmpproduct = new Product($this->db);
 			$result = $tmpproduct->fetch($line->fk_product);
-			foreach($tmpproduct->array_options as $key=>$label)
-				$resarray["line_".$key] = $label;
+			$resarray = $this->fill_substitutionarray_with_extrafields($tmpproduct,$resarray,$extrafields,$array_key="line",$outputlangs);
```

Comment:
Commit 2aaf3a466d9 fixed https://github.com/Dolibarr/dolibarr/issues/3185
But since it didn't call fill_substitutionarray_with_extrafields all field types were not properly expanded

Reply:
souldn't we replace the "line" prefix here with "line_product" to avoid name conflicts between line extrafields and product extrafields ?

Reply:
Yes, we should use "line_product" to avoid conflict.